src/protocol.rs

```diff
@@ -407,6 +416,132 @@ mod tests {
     use std::collections::HashMap;
     use tracing::debug;
 
+    //#[cfg_attr(feature = "flame_it", flame)]
```

Comment:
nit: remove commented code. This line is used for generating the flamegraph for benchmarking. We don't need it on any tests other than the one that runs the whole protocol.